frontend/lib/search/filter_bar_button.dart

```diff
@@ -59,10 +60,24 @@ class _FilterBarButtonState extends State<FilterBarButton>
     const paddingPerElement = 8;
     const minNumberElements = 5;
     var totalWidth = MediaQuery.of(context).size.width;
-    var width = min(
-        80.0,
-        (totalWidth - minNumberElements * paddingPerElement) /
-            minNumberElements);
+
+    // Width allowing at least minNumberElements in one row
+    var smallWidth = min(
+      80.0,
+      (totalWidth - minNumberElements * paddingPerElement) / minNumberElements);
+
+    // At least 65 px are needed to fit all category names
+    var largeWidth = max(
+      65.0,
+      smallWidth);
+
+    var numLargeElementsPerRow = totalWidth ~/ largeWidth;
+    var isSecondRow = widget.index >= max(minNumberElements, numLargeElementsPerRow);
```

Comment:
On a tablet there is no second row, so would this then be always false?

Reply:
Yes